### PR TITLE
Replace custom Neo4j container with testcontainers version.

### PR DESCRIPTION
**Pull request by michael-simons**
We contributed a Neo4j Testcontainers to the project and in this PR I'd like to propose to use that one instead of a custom one. It's basically a drop-in replacement.

### CODE REVIEW

File: spring-boot-project/spring-boot-test-autoconfigure/src/test/java/org/springframework/boot/test/autoconfigure/data/neo4j/DataNeo4jTestIntegrationTests.java [hunk 1]
```diff
@@ -39,14 +39,15 @@
...
+ * @author Michael Simons
  */
 @RunWith(SpringRunner.class)
 @ContextConfiguration(initializers = DataNeo4jTestIntegrationTests.Initializer.class)
 @DataNeo4jTest
 public class DataNeo4jTestIntegrationTests {
 
 	@ClassRule
-	public static Neo4jContainer neo4j = new Neo4jContainer();
+	public static Neo4jContainer neo4j = new Neo4jContainer().withAdminPassword(null);
```
[the remark this answers is not in the document]
Comment (michael-simons) [reply spring-boot-project/spring-boot-test-autoconfigure/src/test/java/org/springframework/boot/test/autoconfigure/data/neo4j/DataNeo4jTestIntegrationTests.java]: Bad API design on my site. I already taking a note to provide a `.disableAuth` or similar and not doing it with a magic null constant.

File: spring-boot-project/spring-boot-test-autoconfigure/src/test/java/org/springframework/boot/test/autoconfigure/data/neo4j/DataNeo4jTestWithIncludeFilterIntegrationTests.java [hunk 2]
```diff
@@ -36,14 +36,15 @@
  * Integration test with custom include filter for {@link DataNeo4jTest}.
  *
  * @author Eddú Meléndez
+ * @author Michael Simons
```
[the remark this answers is not in the document]
Comment (michael-simons) [reply spring-boot-project/spring-boot-test-autoconfigure/src/test/java/org/springframework/boot/test/autoconfigure/data/neo4j/DataNeo4jTestWithIncludeFilterIntegrationTests.java]: Ok, I should have read the updated guidelines. I'll remember for next time.

**Comment by snicoll** [comment]
@michael-simons we've upgraded in #15893 - Can you please rebase and adapt the PR if necessary?

**Comment by wilkinsona** [comment]
@michael-simons No need to rebase. I've already done so locally.

**Comment by wilkinsona** [comment]
Thanks for the PR, @michael-simons. This has been merged into 2.1.x and forwards into master.

**Comment by wilkinsona** [comment]
Re-opening as it looks like the changes are incomplete.

**Pull request closed without merging**